rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -304,18 +416,25 @@ void SequentialCompressionWriter::write(
   metadata_.duration = std::max(metadata_.duration, duration);
 
   auto converted_message = converter_ ? converter_->convert(message) : message;
-  if (compression_options_.compression_mode == rosbag2_compression::CompressionMode::MESSAGE) {
-    compress_message(converted_message);
+  if (compression_options_.compression_mode == CompressionMode::MESSAGE) {
+    std::lock_guard<std::mutex> lock(compressor_mutex_);
+    while (compressor_message_queue_.size() > compression_options_.compression_queue_size) {
+      compressor_message_queue_.pop();
+    }
+    compressor_message_queue_.push(converted_message);
+    compressor_condition_.notify_one();
+  } else {
+    std::lock_guard<std::recursive_mutex> lock(storage_mutex_);
+    storage_->write(converted_message);
   }
-
-  storage_->write(converted_message);
 }
 
-bool SequentialCompressionWriter::should_split_bagfile() const
+bool SequentialCompressionWriter::should_split_bagfile()
```

Comment:
Why remove the `const`?

Reply:
Since multiple threads could be accessing `storage_` at once now, it needs to be protected by a mutex; this method needs to lock that mutex, so it has to be able to modify a class member.